### PR TITLE
pyyaml 3.12 (New Formula)

### PR DESCRIPTION
**Pull request by s172262**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

File: Formula/pyyaml.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  homepage "https://pyyaml.org/wiki/PyYAML"
+  url "https://pyyaml.org/download/pyyaml/PyYAML-3.12.tar.gz"
+  sha256 "592766c6303207a20efc445587778322d7f73b161bd994f227adaa341ba212ab"
+
+  depends_on :python
+  depends_on "libyaml"
+
+  def install
+    args = %W[--prefix=#{prefix}]
+    system "python", "setup.py", "install", *args
```
Comment (ilovezfs) [inline Formula/pyyaml.rb]: should use Python.setup_install_args

File: Formula/pyyaml.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class Pyyaml < Formula
+  desc "YAML parser and emitter for Python"
+  homepage "https://pyyaml.org/wiki/PyYAML"
+  url "https://pyyaml.org/download/pyyaml/PyYAML-3.12.tar.gz"
+  sha256 "592766c6303207a20efc445587778322d7f73b161bd994f227adaa341ba212ab"
+
+  depends_on :python
```
Comment (ilovezfs) [inline Formula/pyyaml.rb]: depends_on :python if MacOS.version <= :snow_leopard

File: Formula/pyyaml.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  depends_on "libyaml"
```
Comment (ilovezfs) [inline Formula/pyyaml.rb]: alphabetize

File: Formula/pyyaml.rb [hunk 4]
```diff
@@ -4,12 +4,11 @@ class Pyyaml < Formula
...
   sha256 "592766c6303207a20efc445587778322d7f73b161bd994f227adaa341ba212ab"
 
-  depends_on :python
   depends_on "libyaml"
+  depends_on :python if MacOS.version <= :snow_leopard
 
   def install
-    args = %W[--prefix=#{prefix}]
-    system "python", "setup.py", "install", *args
+    system "python", *Language::Python.setup_install_args(prefix)
```
Comment (ilovezfs) [inline Formula/pyyaml.rb]: probably
```
    ENV.prepend_create_path "PYTHONPATH", libexec/"lib/python2.7/site-packages"
    system "python", *Language::Python.setup_install_args(libexec)
    bin.install Dir[libexec/"bin/*"]
    bin.env_script_all_files(libexec/"bin", :PYTHONPATH => ENV["PYTHONPATH"])
```

Comment (s172262) [reply Formula/pyyaml.rb]: Ther is no executable file.
```
$ brew list pyyaml
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/ (39 files)

$ brew list pyyaml --verbose
/usr/local/Cellar/pyyaml/3.12/INSTALL_RECEIPT.json
/usr/local/Cellar/pyyaml/3.12/LICENSE
/usr/local/Cellar/pyyaml/3.12/.brew/pyyaml.rb
/usr/local/Cellar/pyyaml/3.12/CHANGES
/usr/local/Cellar/pyyaml/3.12/README
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/PyYAML-3.12-py2.7.egg-info/PKG-INFO
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/PyYAML-3.12-py2.7.egg-info/SOURCES.txt
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/PyYAML-3.12-py2.7.egg-info/top_level.txt
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/PyYAML-3.12-py2.7.egg-info/dependency_links.txt
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/parser.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/cyaml.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/tokens.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/scanner.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/resolver.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/error.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/serializer.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/loader.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/constructor.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/composer.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/events.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/reader.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/events.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/__init__.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/representer.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/tokens.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/dumper.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/cyaml.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/representer.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/constructor.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/parser.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/reader.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/scanner.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/dumper.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/loader.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/error.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/resolver.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/__init__.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/serializer.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/composer.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/nodes.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/nodes.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/emitter.py
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/yaml/emitter.pyc
/usr/local/Cellar/pyyaml/3.12/lib/python2.7/site-packages/_yaml.so
```

**Comment by ilovezfs** [comment]
Ah OK. We only really package Python applications, not Python libraries. (There are a few exceptions (e.g. numpy) for things that take a while to build and are used by other formulae.)

**Comment by s172262** [comment]
I don't want two `libyaml.dylib`s.

**Pull request closed without merging**